### ISSUE TITLE
Short urls http 400 with YOURLS_UNIQUE_URLS=true

### ISSUE THREAD

**Issue by janne-mc** [YOURLS/YOURLS#3547]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://docs.yourls.org/guide/troubleshooting/first-steps.html) and [frequent issues](https://docs.yourls.org/guide/troubleshooting/frequent-issues.html) guides.

### Version

1.9.2

### Description

When using the short url api /yourls-api.php?action=shorturl&url=... with YOURLS_UNIQUE_URLS=true the API request works fine on the first time. On the second and consecutive requests for the same url the short url is returned ok but the http response code is 400. 

I tried also with version 1.8.2 and with that it works as expected.

Could be related to the changes in [#3355](https://github.com/YOURLS/YOURLS/pull/3355)

### Expectation

HTTP reponse code 200

**Comment by dgw** [comment]
The description of #3355, which you referenced, seems to indicate that this behavior is intentional. Are you expecting a 200 Success status code for a request that did not in fact succeed?

**Comment by holll** [comment]
> The description of #3355, which you referenced, seems to indicate that this behavior is intentional. Are you expecting a 200 Success status code for a request that did not in fact succeed?

Yes,According to the HTTP protocol specifications, when the HTTP status code is 400 (Bad Request), the server should not include any body in the response. This is because the 400 status code indicates that the client's request is incorrect or cannot be handled, and the server cannot provide a suitable response.

Please refer to section 6.5.1 of RFC 7231 (https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.1) for details:

> 6.5.1.  400 Bad Request
>
> The 400 (Bad Request) status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).

**Comment by dgw** [comment]
Maybe 409 (Conflict) would be a better response code for this case?

> The request could not be completed due to a conflict with the current state of the resource. This code is only allowed in situations where it is expected that the user might be able to resolve the conflict and resubmit the request. The response body SHOULD include enough information for the user to recognize the source of the conflict. Ideally, the response entity would include enough information for the user or user agent to fix the problem […]
>
> — [RFC 2616 § 10.4.10][1]

[1]: https://datatracker.ietf.org/doc/html/rfc2616#section-10.4.10

The response entity here in fact does contain enough information to identify the source of the conflict, and for the user-agent to fix the error if desired.

**Comment by holll** [comment]
> Maybe 409 (Conflict) would be a better response code for this case?
> 
> > The request could not be completed due to a conflict with the current state of the resource. This code is only allowed in situations where it is expected that the user might be able to resolve the conflict and resubmit the request. The response body SHOULD include enough information for the user to recognize the source of the conflict. Ideally, the response entity would include enough information for the user or user agent to fix the problem […]
> > — [RFC 2616 § 10.4.10](https://datatracker.ietf.org/doc/html/rfc2616#section-10.4.10)
> 
> The response entity here in fact does contain enough information to identify the source of the conflict, and for the user-agent to fix the error if desired.
> 
> (Quoted RFC 2616 because it came up first in search results, but 7231 contains [almost the same](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.8) definition for 409.)

In fact, I found this problem when using php to call this api interface, because the status code is 400, so even if there is content in the response body, the curl function of php thinks that the request failed, and the response body will not be returned by default. But since I'm not proficient in php, I don't know what happens with the 409 status code, maybe you need to test it.

**Comment by flaviovs** [comment]
One suggestion is to add an `existing=true` API parameter that, when passed, would make the call succeed if the URL already exists (current short URL being returned). This would help clients that don't care if the URL already is in the database or not, but also don't want to set `YOURLS_UNIQUE_URLS=false`.

**Comment by wgicio** [comment]
N8n throws an error on the node when the 400 code is returned, would be good to have this fixed up

**Comment by dgw** [comment]
@wgicio 30 seconds on google found https://community.n8n.io/t/http-node-error-400-not-handled/9739

It is possible to modify a workflow on n8n to handle HTTP 400.

**Comment by bh-elder** [comment]
I created a plugin to resolve this problem at my company -- it makes YOURLS effectively treat duplicate links as success (returning the link that we already have in the database) rather than bad requests. It seems to work well for our use case, but as with all OSS if it breaks you get to keep both parts 😁 https://github.com/elder-oss/yourls-allow-existing-urls
